sct-commons/src/main/java/org/lfenergy/compas/sct/commons/LnodeTypeService.java

```diff
@@ -25,4 +28,32 @@ public Optional<TLNodeType> findLnodeType(TDataTypeTemplates tDataTypeTemplates,
         return getFilteredLnodeTypes(tDataTypeTemplates, tlNodeTypePredicate).findFirst();
     }
 
+    public Stream<DataAttributeRef> getAllDOAndDA(TDataTypeTemplates dtt, TLNodeType tlNodeType, DataAttributeRef dataRef)  {
```

Comment:
Why this method is in `LnodeTypeService` ? since it does not return any `TLNodeType`

Reply:
> is the method name appropriated ?

Here we are collecting a list of data that represents all DO and DA objects (I didn't mention subdata like SDO or BDA in the name, because it is unnecessary)
I think the name is appropriated.

Reply:
> Why this method is in `LnodeTypeService` ? since it does not return any `TLNodeType`

I considered `DataAttributeRef` as our domain object, it refers to a structured data that can be found by exploring a number of objects, the entry object is TLNodeType. I didn't rename `DataAttributeRef` but we can think of it as `LNodeTypeDomain`. 
The reason why this method is included in `LnodeTypeService` because it returns an object that closely resembles LNodeType. 

For more details you may look at this `LNodeTypeAdapter#getDataAttributeRefs`

Reply:
> > is the method name appropriated ?
> 
> Here we are collecting a list of data that represents all DO and DA objects (I didn't mention subdata like SDO or BDA in the name, because it is unnecessary) I think the name is appropriated.

if you return DO and DA, please use another return object

Reply:
`getAllDoLinkedToDa`